chaco/tools/tests/test_range_selection.py

```diff
@@ -65,7 +61,12 @@ def test_selection_no_warning(self):
         renderer = plot.plot(('x', 'y'))[0]
         tool = RangeSelection(renderer)
         with warnings.catch_warnings(record=True) as w:
+            # Ignore warnings coming from Traits
+            warnings.filterwarnings(
```

Comment:
Being a bit picky here, but I think this is changing the global state of the warning filters but not resetting it, and so could affect the outcome of other tests which mess with warnings.  Unfortunately this looks awkward to do in the warnings module withot messing with `warnings.filter` directly.

Reply:
I don't think this is an issue here, since it's wrapped in a `catch_warnings` block which will reset the filters upon exit:
```python
In [1]: import warnings

In [2]: print(warnings.filters)
[('default', re.compile('', re.IGNORECASE), <class 'DeprecationWarning'>, re.compile('__main__'), 0), ('ignore', None, <class 'DeprecationWarning'>, None, 0), ('ignore', None, <class 'PendingDeprecationWarning'>, None, 0), ('ignore', None, <class 'ImportWarning'>, None, 0), ('ignore', None, <class 'BytesWarning'>, None, 0), ('ignore', None, <class 'ResourceWarning'>, None, 0)]

In [3]: with warnings.catch_warnings(record=True) as w:
   ...:     warnings.filterwarnings("ignore", 'elementwise == comparison failed')
   ...:     print(warnings.filters)
   ...:     
[('ignore', re.compile('elementwise == comparison failed', re.IGNORECASE), <class 'Warning'>, re.compile(''), 0), ('default', re.compile('', re.IGNORECASE), <class 'DeprecationWarning'>, re.compile('__main__'), 0), ('ignore', None, <class 'DeprecationWarning'>, None, 0), ('ignore', None, <class 'PendingDeprecationWarning'>, None, 0), ('ignore', None, <class 'ImportWarning'>, None, 0), ('ignore', None, <class 'BytesWarning'>, None, 0), ('ignore', None, <class 'ResourceWarning'>, None, 0)]

In [4]: print(warnings.filters)
[('default', re.compile('', re.IGNORECASE), <class 'DeprecationWarning'>, re.compile('__main__'), 0), ('ignore', None, <class 'DeprecationWarning'>, None, 0), ('ignore', None, <class 'PendingDeprecationWarning'>, None, 0), ('ignore', None, <class 'ImportWarning'>, None, 0), ('ignore', None, <class 'BytesWarning'>, None, 0), ('ignore', None, <class 'ResourceWarning'>, None, 0)]
```

Reply:
Cool, I didn't realise that.